ignite/distributed/launcher.py

```diff
@@ -0,0 +1,265 @@
+
+        Single node option :
+
+        .. code-block:: bash
+
+            python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py
+
+        Multi-node option :
+
+        .. code-block:: bash
+
+            # node 0
+            python -m torch.distributed.launch --nnodes=2 --node_rank=0 --master_addr=master \
+                --master_port=3344 --nproc_per_node=8 --use_env main.py
+
+            # node 1
+            python -m torch.distributed.launch --nnodes=2 --node_rank=1 --master_addr=master \
+                --master_port=3344 --nproc_per_node=8 --use_env main.py
+
+
+        User code is the same for both options:
+
+        .. code-block:: python
+
+            # main.py
+
+            import ignite.distributed as idist
+
+            def training(local_rank, config, **kwargs):
+                # ...
+                print(idist.get_rank(), ": run with config:", config, "- backend=", idist.backend())
+                # ...
+
+            with idist.Parallel(backend="nccl") as parallel:
+                parallel.run(training, config, a=1, b=2)
+
+
+        2) Single node, Multi-GPU training launched with `python`
+
+        .. code-block:: bash
+
+            python main.py
+
+        .. code-block:: python
+
+            # main.py
+
+            import ignite.distributed as idist
+
+            def training(local_rank, config, **kwargs):
+                # ...
+                print(idist.get_rank(), ": run with config:", config, "- backend=", idist.backend())
+                # ...
+
+            with idist.Parallel(backend="nccl", num_procs_per_node=4) as parallel:
+                parallel.run(training, config, a=1, b=2)
+
+
+        3) Single node, Multi-TPU training launched with `python`
+
+        .. code-block:: bash
+
+            python main.py
+
+        .. code-block:: python
+
+            # main.py
+
+            import ignite.distributed as idist
+
+            def training(local_rank, config, **kwargs):
+                # ...
+                print(idist.get_rank(), ": run with config:", config, "- backend=", idist.backend())
+                # ...
+
+            with idist.Parallel(backend="xla-tpu", num_procs_per_node=8) as parallel:
+                parallel.run(training, config, a=1, b=2)
+
+
+        4) Multi-node, Multi-GPU training launched with `python`. For example, 2 nodes with 8 GPUs:
+
+        .. code-block:: bash
+
+            # node 0
+            python main.py --node_rank=0
+
+            # node 1
+            python main.py --node_rank=1
+
+
+        .. code-block:: python
+
+            # main.py
+
+            import ignite.distributed as idist
+
+            def training(local_rank, config, **kwargs):
+                # ...
+                print(idist.get_rank(), ": run with config:", config, "- backend=", idist.backend())
+                # ...
+
+            dist_config = {
```

Comment:
Why do not provide `torch.distributed.launch` API ? I mean `num_procs_per_node` => `nproc_per_node` etc.

Reply:
This is because of `idist.spawn` args. Maybe we could make it more uniform between `num_procs_per_node`, `nproc_per_node` and even `idist.get_ntasks_per_node()`